src/MigrationTools.Clients.AzureDevops.ObjectModel/Enrichers/TfsEmbededImagesEnricher.cs

```diff
@@ -48,71 +51,99 @@ public override int Enrich(WorkItemData sourceWorkItem, WorkItemData targetWorkI
                     {
-                        if (match.Value.ToLower().Contains(oldTfsurl.ToLower()) || match.Value.ToLower().Contains(oldTfsurlOppositeSchema.ToLower()))
+                        if (!match.Value.ToLower().Contains(oldTfsurl.ToLower()) && !match.Value.ToLower().Contains(oldTfsurlOppositeSchema.ToLower()))
+                            continue;
+
+                        //save image locally and upload as attachment
+                        Match newFileNameMatch = Regex.Match(match.Value, RegexPatternForImageFileName, RegexOptions.IgnoreCase);
+                        if (!newFileNameMatch.Success)
+                            continue;
+
+                        Log.LogDebug($"{logTypeName}: field '{field.Name}' has match: {WebUtility.HtmlDecode(match.Value)}");
+                        string fullImageFilePath = Path.GetTempPath() + newFileNameMatch.Value;
+
+                        try
                         {
-                            //save image locally and upload as attachment
-                            Match newFileNameMatch = Regex.Match(match.Value, regExSearchFileName, RegexOptions.IgnoreCase);
-                            if (newFileNameMatch.Success)
+                            using (var httpClient = new HttpClient(_httpClientHandler, false))
                             {
-                                Log.LogDebug("EmbededImagesRepairEnricher: field '{fieldName}' has match: {matchValue}", field.Name, System.Net.WebUtility.HtmlDecode(match.Value));
-                                string fullImageFilePath = Path.GetTempPath() + newFileNameMatch.Value;
-
-                                using (var httpClient = new HttpClient(_httpClientHandler, false))
+                                if (!string.IsNullOrEmpty(sourcePersonalAccessToken))
                                 {
-                                    if (!string.IsNullOrEmpty(sourcePersonalAccessToken))
+                                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", "", sourcePersonalAccessToken))));
+                                }
+                                var result = DownloadFile(httpClient, match.Value, fullImageFilePath);
+                                if (!result.IsSuccessStatusCode)
+                                {
+                                    if (_ignore404Errors && result.StatusCode == HttpStatusCode.NotFound)
                                     {
-                                        httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes(string.Format("{0}:{1}", "", sourcePersonalAccessToken))));
+                                        Log.LogDebug($"{logTypeName}: Image {match.Value} could not be found in WorkItem {wi.Id}, Field {field.Name}");
```

Comment:
Please return all "LogInformation" lines to the format `"test {item} and {item2}", item, itesm2` format. This is a deliberate setup using a logging platform that converts variables to telemetry and hilights.

Reply:
Updated as requested..